### PR TITLE
Minor generalizations

### PR DESCRIPTION
**Pull request by jarretlavallee**
Some minor things we discussed in #2 to allow the tasks to be more generic.

### CODE REVIEW

File: tasks/pin_node_group.rb [hunk 1]
```diff
@@ -19,10 +23,11 @@
 
 # Get the id of the 'PE Master' node group
```
Comment (m0dular) [inline tasks/pin_node_group.rb]: This comment and the one at 29 need to be changed now that these aren't specific to `PE Master`.

File: tasks/run_agent.sh [hunk 2]
```diff
@@ -4,12 +4,12 @@
 declare PT__installdir
 source "$PT__installdir/deploy_pe/files/common.sh"
 PUPPET_BIN=/opt/puppetlabs/bin
+retries=${PT_retries:-5}
```
Comment (m0dular) [inline tasks/run_agent.sh]: I wonder why shellcheck doesn't flag this as needing to be quoted?  I guess it's ok in this context.

**Review by m0dular — CHANGES_REQUESTED** [review]
Good idea, and everything looks good except for the two comments I pointed out.  Thanks!

File: tasks/run_agent.sh (hunk 2 above)
Comment (jarretlavallee) [reply tasks/run_agent.sh]: I think it is because if it does not exist, we are setting the default.

File: tasks/pin_node_group.rb (hunk 1 above)
Comment (jarretlavallee) [reply tasks/pin_node_group.rb]: good call

File: tasks/run_agent.sh [hunk 3]
```diff
@@ -4,12 +4,12 @@
...
+wait_time=${PT_retries:-300}
```
Comment (jarretlavallee) [inline tasks/run_agent.sh]: This was using the wrong variable.

**Comment by jarretlavallee** [comment]
@m0dular If it looks good after the changes, feel free to merge it or let me know what else I can/should change.

File: tasks/pin_node_group.rb [hunk 4]
```diff
@@ -17,12 +21,13 @@
...
 http.ca_file = Puppet.settings['localcacert']
 
-# Get the id of the 'PE Master' node group
+# Get the id of the node group
 res = http.request_get(uri.request_uri).read_body
-master_id = JSON.parse(res).find { |group| group['name'] == 'PE Master' }['id']
+node_group_id = JSON.parse(res).find { |group| group['name'] == node_group }['id']
+raise "No group ID could be found for #{node_group}" if node_group_id.nil?
 
 # use the /pin endpoint to pin our compile masters to PE Master
```
Comment (m0dular) [inline tasks/pin_node_group.rb]: Can you also generalize this comment?

**Review by m0dular — CHANGES_REQUESTED** [review]

File: tasks/run_agent.sh (hunk 2 above)
Comment (m0dular) [inline tasks/run_agent.sh]: One of the things `common.sh` does is to munge the `PT_` variables so you can reference them without the prefix.  Can you change these two lines to:

```
retries="${retries:-5}"
wait_time="${wait_time:-300}"
```

File: tasks/pin_node_group.rb (hunk 4 above)
Comment (jarretlavallee) [reply tasks/pin_node_group.rb]: @m0dular I finally got back to this. Let me know if that works for you.

**Review by m0dular — COMMENTED** [review]
Thanks, the comments look good now.  Can you make the change I pointed out on lines 7-8 about munging `PT_` variables?  Then it should be good to go.

File: tasks/run_agent.sh (hunk 2 above)
Comment (jarretlavallee) [reply tasks/run_agent.sh]: Done.

**Pull request merged**